Automattic/sensei
# Integrate extensions page with real data

### Changes proposed in this Pull Request

* Introduces a new REST endpoint point `sensei-plugins` which is used by the extensions page.
* Switch fronted to use the new edpoint.
* Various fixes after integrating with real data.
* There is definitely more work/testing needed but I decided to raise the PR before pausing.
* There is missing functionality (handling updates, connecting to WC.com etc) which is going to be handled as part of another PR.
* We also need to specify the categories which every plugin falls into (i.e. featured, course creation, learner engagment)

### Testing instructions

* Visit the extensions page and observe that the content is generated with real data.
* As it is now there are no plugins categorised as 'Featured',  'Course creation' and 'Learner engagment' so this list will either contain every plugin or none at all. 
* You can modify the main plugin files of each plugin to test various scenarios.
